python/ray/monitor.py

```diff
@@ -205,11 +203,11 @@ def cleanup_task_table(self):
                 ray.experimental.state.TASK_STATUS_LOST, NIL_ID,
                 task["ExecutionDependenciesString"], task["SpillbackCount"])
             if ok != b"OK":
-                log.warn("Failed to update lost task for dead scheduler.")
+                logger.warn("Failed to update lost task for dead scheduler.")
             num_tasks_updated += 1
 
         if num_tasks_updated > 0:
-            log.warn("Marked {} tasks as lost.".format(num_tasks_updated))
+            logger.warn("Marked {} tasks as lost.".format(num_tasks_updated))
```

Comment:
I just now noticed this and maybe this does not have to be part of this PR, but if you are already fixing all this. Ideally, one would not format logging strings before passing them to logger, but would leave it to logger to format them. In this way you can get arguments provided outside of strings, which can help when you maybe want to translate logging strings and so on.

So the syntax would then be:

```
logger.warn("Market %s tasks as lost.", num_tasks_updated)
```

Or, I personally prefer:

```
logger.warn("Market %(num_tasks_updated)s tasks as lost.", {'num_tasks_updated': num_tasks_updated})
```

Those arguments are then available as `args` in the logging record.

Sadly, you have to use the old-style interpolation and not new Python formatting.

Reply:
I think Robert would be against this change at this time. He wants the string format uses the same style. I think in this PR we can focus on changing the global logger to module-wise logger and make the logger configurable. And you can open another PR to do this format change. ;-)

Reply:
@robertnishihara Here the idea is that we keep strings separate. That we do not pass to `warn` an already formatted string. This allows interpolation later on.

I do agree that `{}` format is nicer when you want to format strings in your code, but this can be seen as simply passing arguments to a function in the format it expects.

Reply:
I don't feel *that* strongly about how we format strings. However, I do object to changing it in some files and not others.

We do a lot of string formatting outside of logging statements, so I think it'll be simpler (and easier to review) if we just leave it alone for now.

Reply:
I think the best is to see this just as a different API. Not formatting but providing arguments in the way the call requires it.

See also those two linting messages:
* http://pylint-messages.wikidot.com/messages:w1201
* https://stackoverflow.com/questions/29147442/how-to-fix-pylint-logging-not-lazy

By leaving arguments out, you are not unnecessary formatting strings, if the logging level will not even pass those messages. For example, for debugging logging statements which might be issued very often, this might slow down things.